site/book/03-packages/05-updating-a-package.md

```diff
@@ -65,4 +65,97 @@ Once you have successfully updated the package, commit the changes:
 $ git add .; git commit -m "Updated wordpress to v0.4"
 ```
 
+## Merge strategies
+
+### Resource-merge
+
+The `resource-merge` strategy performs a structural comparison of each resource using the
+OpenAPI schema. So rather than performing a text-based merge, `kpt` leverages the
+common structure of KRM resources.
+
+#### Resource identity
+In order to perform a per-resource merge, `kpt` needs to be able to match a resource in
+the local package with the same resource in the upstream version of the package. It does
+this matching based on the identity of a resource, which is the combination of group,
+kind, name and namespace. So in our `wordpress` example, the identity of the`Deployment`
+resource is:
+```
+group: apps
+kind: Deployment
+name: wordpress
+namespace: ""
+```
+Changing the name and/or namespace of a resource is a pretty common way to customize
+a package. In order to make sure this doesn't create problems during merge, `kpt` will
+automatically adding the `# kpt-merge: <namespace>/<name>` comment on the `metadata` 
+field of every resource when getting or updating a package. An example is the `Deployment`
+resource from the `wordpress` package:
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata: # kpt-merge: /wordpress
+  name: wordpress
+  labels:
+    app: wordpress
+...
+```
+
+#### Merge rules
+`kpt` performs a 3-way merge for every resource. This means it will use the resource
+in the local package, the updated resource from upstream, as well as the resource
+at the version where the local and upstream package diverged (i.e.
+common ancestor). When discussing the merge rules in detail, we will be referring to
+the three different sources as local, upstream and origin.
+
+In the discussion, we will be referring to non-associative and associative lists. A
+non-associative list either has elements that are scalars or another list, or it has elements
+that are mappings but without an associative key. An associative list has elements that are mappings and
+one or more of the fields in the mappings are designated as associative keys. An associative key
+(also sometimes referred to as a merge key) is used to identify the "same" elements in two
+different lists for the purpose of merging them. `kpt` will primarily look for information about
+any associative keys from the OpenAPI schema, but some fields are also automatically recognized as
+associative keys:
+* `mountPath`
+* `devicePath`
+* `ip`
+* `type`
+* `topologyKey`
+* `name`
+* `containerPort`
+
+On the resource level, the merge rules are as follows:
```

Comment:
This is probably better represented as as a table. Same for other rules below.

Reply:
I tried to turn this into tables. I'm not sure how it works.